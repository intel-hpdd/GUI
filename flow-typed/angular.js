// @flow

declare module angular {
  declare class $scope {
    $new():$scope;
    $digest():void;
  }
  declare type $scopeT = $scope;
  declare class Module {
    provider(name:string, providerType:Function):Module;
    factory(name:string, providerFunction:Function):Module;
    controller(name:string, constructor:Function):Module;
    service(name:string, constructor:Function):Module;
    config(configFn:Function):Module;
    run(initializationFn:Function):Module;
    directive(name:string, directiveFactory:Function):Module;
    component(name:string, options:Object):Module;
    filter(name:string, filterFactory:Function):Module;
    value(name:string, object:any):Module;
    constant(name:string, object:any):Module;
    name:string;
  }
  declare type $compileT = (el:(string | HTMLElement)) => (scope:$scope) => HTMLElement[];
  declare module.exports: {
    module(name:string, dependencies: string[]):Module;
    bootstrap(element:(HTMLElement | Document), modules:string[], config:Object):void;
    merge(...rest:Object[]):Object;
    element(el:(string | HTMLElement)): {
      clone: () => HTMLElement
    };
  }
}
