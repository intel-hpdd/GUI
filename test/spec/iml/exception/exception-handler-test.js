import exceptionHandlerConfig from '../../../../source/iml/exception/exception-handler.js';
import windowUnloadFactory from '../../../../source/iml/window-unload.js';
import angular from '../../../angular-mock-setup.js';

describe('exception handler', () => {
  let oldExceptionHandler;

  beforeEach(
    angular.mock.module(($provide, $exceptionHandlerProvider) => {
      $exceptionHandlerProvider.mode('log');
      oldExceptionHandler = $exceptionHandlerProvider.$get();

      $provide.factory('windowUnload', windowUnloadFactory);
      $provide.value('exceptionModal', jest.fn());
      exceptionHandlerConfig($provide);
    })
  );

  let $exceptionHandler, exceptionModal, windowUnload, error, cause;

  beforeEach(
    angular.mock.inject((_$exceptionHandler_, _exceptionModal_, _windowUnload_) => {
      error = new Error('uh oh!');
      cause = 'Something Happened!';

      $exceptionHandler = _$exceptionHandler_;
      exceptionModal = _exceptionModal_;
      windowUnload = _windowUnload_;
    })
  );

  afterEach(() => {
    windowUnload.unloading = false;
  });

  it('should not open the modal if the window is unloading', () => {
    windowUnload.unloading = true;
    $exceptionHandler(new Error('foo'), 'bar');

    expect(exceptionModal).not.toHaveBeenCalled();
  });

  describe('handling an exception', () => {
    let oldFetch;

    beforeEach(() => {
      oldFetch = window.fetch;
      window.fetch = jest.fn(() => 'fetch');
      $exceptionHandler(error, cause);
    });

    afterEach(() => {
      window.fetch = oldFetch;
    });

    it('should send the request', () => {
      expect(window.fetch).toHaveBeenCalledOnceWith('/iml-srcmap-reverse', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json; charset=UTF-8'
        },
        body: JSON.stringify({
          trace: error.stack
        })
      });
    });

    it('should open the modal when there is an error', () => {
      expect(exceptionModal).toHaveBeenCalled();
    });

    it('should only open the modal once', () => {
      $exceptionHandler(error, cause);

      expect(exceptionModal).toHaveBeenCalledTimes(1);
    });

    it('should delegate to the older $exceptionHandler', () => {
      expect(oldExceptionHandler.errors[0]).toEqual([error, cause]);
    });
  });
});
