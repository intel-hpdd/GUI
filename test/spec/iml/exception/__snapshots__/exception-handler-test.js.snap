// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exception modal component should generate the modal 1`] = `
<div>
  <div
    class="modal fade in exception-modal"
    role="dialog"
    style="display: block; z-index: 2000;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h3>
            <i
              class="fa fa-exclamation-triangle"
            />
            Encountered An Error
          </h3>
        </div>
        <div
          class="modal-body text-center"
        >
          <p>
            An unexpected error has occurred. 
            <br />
            Please collect iml-diagnostics and send to support.
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-large btn-block"
            type="button"
          >
            <i
              class="fa fa-refresh"
            />
             Reload
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal-backdrop fade in exception-modal-backdrop"
    style="z-index: 1999;"
  />
</div>
`;
